### DISCUSSION TITLE
How does Provider-less mode work under the hood?

### DISCUSSION THREAD

**Discussion by Ephil012** [pmndrs/jotai#2185] [General]
For provider-less mode, how does it work under the hood to prevent state from being lost on navigation/unmount? To handle this with `useContext`, you might use a provider to keep a global state for your app (whether you're in light or dark mode, etc). For instance
```
<ThemeContext.Provider value="dark">
    ...
<ThemeContext.Provider value="dark">
 ```
 If you're using this with say NextJS (or some other library/framework with navigation) then you just put this provider at the top and it will persist even on navigation afaik.
 
 However, after looking at the code for provider-less mode, it seems it just has a `createDefaultStore` function which just calls `createStore` if the store doesn't already exist. That `createStore` function just sets up a `WeakMap` which can have it's keys garbage collected. So let's say I unmount every component that references a jotai atom when I navigate. What will happen to the state? Will it just be garbage collected and lost? This seems less than ideal if you wanted to preserve some state across all pages of your app, even if it's technically not used by the current page you're on. How does jotai prevent the state from being lost in this case? The only thing I could really think of is to use Jotai's `Provider` to mimic how React Providers work so the state doesn't get lost.

**Comment by dai-shi** [comment]
> How does Provider-less mode work under the hood?

It simply uses the React context default value.

```jsx
const ThemeContext = createContext("light");

  <ThemeContext.Provider value="dark">
   // theme is dark here
  </ThemeContext.Provider>
  // theme is light here
```

So, it's not actually a "mode", it's just using a default store. Sorry if the wording confused you.
